Spigot-API-Patches/0201-Add-no-tick-view-distance-API.patch

```diff
@@ -0,0 +1,87 @@
+From 20f4438fc7a3623abd88a00464e253ed379c17c8 Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Spottedleaf@users.noreply.github.com>
+Date: Mon, 27 Apr 2020 14:06:49 +0200
+Subject: [PATCH] Add no-tick view distance API
+
+Originally from: https://github.com/Spottedleaf/Tuinity/blob/master/patches/api/0003-View-distance-api.patch
+Licence: MIT
+
+This allows setting a view distance where entities and chunks as a whole
+do not tick. Servers can use this to make it seem as if they have a
+higher view distance setting than they truly do.
+
+diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
+index db18f70e..ca908995 100644
+--- a/src/main/java/org/bukkit/World.java
++++ b/src/main/java/org/bukkit/World.java
+@@ -3216,6 +3216,32 @@ public interface World extends PluginMessageRecipient, Metadatable {
+     @NotNull
+     public List<Raid> getRaids();
+ 
++    // Paper start - View distance API
++    // Originally: https://github.com/Spottedleaf/Tuinity/blob/master/patches/api/0003-View-distance-api.patch
++    // Credit: Spottedleaf
++    /**
++     * Returns the no-tick view distance for this world.
++     * <p>
++     * No-tick view distance is the view distance where chunks will load,
++     * however the chunks and their entities will not be set to tick.
++     * </p>
++     *
++     * @return The no-tick view distance for this world.
++     */
++    public int getNoTickViewDistance();
++
++    /**
++     * Sets the no-tick view distance for this world.
++     * <p>
++     * No-tick view distance is the view distance where chunks will load,
++     * however the chunks and their entities will not be set to tick.
++     * </p>
++     *
++     * @param viewDistance The new view distance in [2, 32].
++     */
++    public void setNoTickViewDistance(int viewDistance);
++    // Paper end
++
+     /**
+      * Get the {@link DragonBattle} associated with this world.
+      *
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 787bb144..9a9fe4c0 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1830,4 +1830,30 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     @Override
+     Spigot spigot();
+     // Spigot end
++
++    // Paper start - View distance API
++    // Originally: https://github.com/Spottedleaf/Tuinity/blob/master/patches/api/0003-View-distance-api.patch
++    // Credit: Spottedleaf
++    /**
++     * Returns the no-tick view distance for this player.
++     * <p>
++     * No-tick view distance is the view distance where chunks will load,
++     * however the chunks and their entities will not be set to tick.
++     * </p>
++     *
++     * @return The no-tick view distance for this world.
```

Comment:
Copy and paste mistake? Seems like it should be `for this player`

Reply:
Indeed it is! I will fix this shortly